pkg/api/customization/cluster/actionhandler.go

```diff
@@ -525,3 +531,65 @@ func (a ActionHandler) RotateCertificates(actionName string, action *types.Actio
 	http.ServeContent(apiContext.Response, apiContext.Request, "rotate", time.Now(), reader)
 	return nil
 }
+
+func (a ActionHandler) SaveSnapshotHandeler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	cluster, err := a.ClusterClient.Get(apiContext.ID, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	newBackup := etcdbackup.NewBackupObject(cluster)
+	response := map[string]interface{}{
+		"snapshot": newBackup.Name,
+	}
+	if _, err = a.BackupClient.Create(newBackup); err != nil {
+		response["message"] = err.Error()
+		apiContext.WriteResponse(http.StatusInternalServerError, response)
+		return err
+	}
+	response["message"] = "Success"
+	apiContext.WriteResponse(http.StatusCreated, response)
+	return nil
+}
+
+func (a ActionHandler) RestoreSnapshotHandler(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	data, err := ioutil.ReadAll(apiContext.Request.Body)
+	if err != nil {
+		return errors.Wrap(err, "reading request body error")
+	}
+
+	input := mgmtclient.RestoreSnapshotInput{}
+	if err = json.Unmarshal(data, &input); err != nil {
+		return errors.Wrap(err, "unmarshaling input error")
+	}
+	response := map[string]interface{}{
+		"snapshot": input.SnapshotID,
+	}
+
+	cluster, err := a.ClusterClient.Get(apiContext.ID, metav1.GetOptions{})
+	if err != nil {
+		response["message"] = err.Error()
+		apiContext.WriteResponse(http.StatusInternalServerError, response)
+		return err
+	}
+
+	if input.BackupTarget == nil {
```

Comment:
if the point is to check the access to the cluster, lines 576-583 should be done prior to getting the cluster from the client (line 569)